File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int maxNestingDepth = 0, currentDepth = 0;
        for (char& ch : s) {
            if (ch == '(') {
                currentDepth++;
                maxNestingDepth = max(maxNestingDepth, currentDepth);
            } else if (ch == ')') {
                currentDepth--;
            }
        }
        return maxNestingDepth;
    }
};

File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        max_nesting_depth = 0
        current_depth = 0
        for ch in s:
            if ch == "(":
                current_depth += 1
                max_nesting_depth = max(max_nesting_depth, current_depth)
            elif ch == ")":
                current_depth -= 1
        return max_nesting_depth
